### PR TITLE
itk: add components

### PR DESCRIPTION
**Pull request by SpaceIm**
- relocatable shared libs on macOS: see https://github.com/conan-io/hooks/issues/376
- bump dependencies
- add components.

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by uilianries — DISMISSED** [review]

**Review by Nekto89 — DISMISSED** [review]

**Review by SSE4 — DISMISSED** [review]

**Review by uilianries — APPROVED** [review]
LGTM

File: recipes/itk/all/patches/0001-find-conan-packages.patch [hunk 1]
```diff
@@ -140,7 +140,7 @@
 -itk_module_target(itkopenjpeg NO_INSTALL)
 +find_package(OpenJPEG REQUIRED)
 +add_library(itkopenjpeg INTERFACE)
-+set_property(TARGET itkopenjpeg PROPERTY INTERFACE_LINK_LIBRARIES OpenJPEG::OpenJPEG)
++set_property(TARGET itkopenjpeg PROPERTY INTERFACE_LINK_LIBRARIES openjp2)
```
Comment (SSE4) [inline recipes/itk/all/patches/0001-find-conan-packages.patch:143]: should we fix target name in openjpeg?

**Review by SSE4 — APPROVED** [review]
wow! that's a lot of components!

File: recipes/itk/all/patches/0001-find-conan-packages.patch (hunk 1 above)
Comment (uilianries) [reply recipes/itk/all/patches/0001-find-conan-packages.patch:143]: Are they incorrect? If yes, I agree changing

Comment (SpaceIm) [reply recipes/itk/all/patches/0001-find-conan-packages.patch:143]: `openjp2` is the official target

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**